### PR TITLE
[kubectl] Don't require port for headless service expose

### LINKED ISSUE #32795: kubectl expose --cluster-ip=None demands a port, not needed
API validation explicitly allows headless services to not have ports, but we seem to have not updated kubectl to know this:

https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation.go#L2390

### PR DESCRIPTION
**Pull request by nebril**
The validation for this is already done in API, but kubectl didn't allow that.
fixes #32795

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36434)

### CODE REVIEW

**Comment by smarterclayton** [comment]
Please add a test case to verify the new behavior.

**Comment by nebril** [comment]
@smarterclayton done

**Comment by nebril** [comment]
@smarterclayton pinging for review

File: pkg/kubectl/cmd/expose_test.go [hunk 1]
```diff
@@ -429,6 +429,38 @@ func TestRunExposeService(t *testing.T) {
...
+				},
+			},
+			flags: map[string]string{"selector": "func=stream", "protocol": "UDP", "name": "foo", "labels": "svc=test", "cluster-ip": "None", "dry-run": "true"},
+			output: &api.Service{
+				ObjectMeta: api.ObjectMeta{Name: "foo", Namespace: "", Labels: map[string]string{"svc": "test"}},
+				Spec: api.ServiceSpec{
+					Ports: []api.ServicePort{
+						{
+							Protocol:   api.ProtocolUDP,
+							Port:       14,
```
Comment (kargakis) [inline pkg/kubectl/cmd/expose_test.go]: Where did this number come from?

**Comment by nebril** [comment]
@smarterclayton pinging for review

File: pkg/kubectl/service.go [hunk 2]
```diff
@@ -136,39 +136,41 @@ func generate(genericParams map[string]interface{}) (runtime.Object, error) {
 		}
```
Comment (kargakis) [inline pkg/kubectl/service.go]: Won't the error above be returned in case there is no port provided?

**Comment by kargakis** [comment]
Can you add an additional test in hack/make-rules/test-cmd.sh?

File: pkg/kubectl/service.go (hunk 2 above)
Comment (nebril) [inline pkg/kubectl/service.go]: It seems I missed it, will check it and fix, but it certainly appears so. Thanks!

**Comment by nebril** [comment]
@kargakis are there any docs on how to run these hack/make-rules/test-cmd.sh tests locally?

**Comment by kargakis** [comment]
make test-cmd

**Comment by kargakis** [comment]
Seems like this is superseded by https://github.com/kubernetes/kubernetes/pull/32811

**Pull request closed without merging**